src/event/event.rs

```diff
@@ -190,16 +190,24 @@ impl Event {
 
 impl fmt::Debug for Event {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("Event")
-            .field("token", &self.token())
+        let alternate = f.alternate();
+        let d = &mut f.debug_struct("Event");
+        d.field("token", &self.token())
             .field("readable", &self.is_readable())
             .field("writable", &self.is_writable())
             .field("error", &self.is_error())
             .field("read_closed", &self.is_read_closed())
             .field("write_closed", &self.is_write_closed())
             .field("priority", &self.is_priority())
             .field("aio", &self.is_aio())
-            .field("lio", &self.is_lio())
-            .finish()
+            .field("lio", &self.is_lio());
+
+        if alternate {
+            let mut data_str = String::new();
+            sys::event::get_platform_event_data_str(&mut data_str, &self.inner);
```

Comment:
Java level function name ;)

Reply:
Maybe call it `details`?

Reply:
renamed to write_details, fwiw I like long explanatory names :)